src/pages/filter/ui/modal-edit/components/modal-compound-request/components/request-block.tsx

```diff
@@ -16,55 +17,77 @@ interface IRequestBlockProps {
-            <div
-              className="step-content-area"
-              onClick={(e: React.MouseEvent) =>
-                modalCompoundRequestStore.setActiveRequest(index, e)
+    useEffect(() => {
+      try {
+        setErrors(Number.parseInt(item[0]) <= 0)
+      } catch (e) {
+        setErrors(true)
+      }
+    }, [item])
+
+    return (
+      <div className={cn('flex flex-col w-full')}>
+        <div
+          className={cn(
+            'flex justify-between w-full shadow-dark py-2 my-2 px-2 cursor-pointer step-content-area',
+            index === activeRequestIndex ? 'bg-blue-medium' : 'bg-white',
+          )}
+          onClick={(e: React.MouseEvent) =>
+            modalCompoundRequestStore.setActiveRequest(index, e)
+          }
+        >
+          <div className="flex cursor-pointer step-content-area">
+            <InputNumber
+              value={item[0]}
+              onChange={(e: ChangeEvent<HTMLInputElement>) =>
+                modalCompoundRequestStore.changeRequestConditionNumber(
+                  index,
+                  e.target.value,
+                )
               }
-              key={group}
-            >
-              <span className="cursor-pointer step-content-area">{group}</span>
+              className="shadow-dark w-1/3 h-5"
+            />
+          </div>
 
-              <Select
-                onChange={(e: ChangeEvent<HTMLSelectElement>) =>
-                  modalCompoundRequestStore.setSingleScenario(
-                    index,
-                    currNo,
-                    e.target,
-                  )
+          <div className="flex flex-1 justify-between step-content-area">
+            {problemGroups.map((group: string, currNo: number) => (
+              <div
+                className="step-content-area"
+                onClick={(e: React.MouseEvent) =>
+                  modalCompoundRequestStore.setActiveRequest(index, e)
                 }
-                className="w-auto ml-1"
-                options={selectOptions}
-                value={modalCompoundRequestStore.getSelectedValue(group, index)}
-              />
-            </div>
-          ))}
+                key={group}
+              >
+                <span className="cursor-pointer step-content-area">
+                  {group}
+                </span>
+
+                <Select
+                  onChange={(e: ChangeEvent<HTMLSelectElement>) =>
+                    modalCompoundRequestStore.setSingleScenario(
+                      index,
+                      currNo,
+                      e.target,
+                    )
+                  }
+                  className="w-auto ml-1"
+                  options={selectOptions}
+                  value={modalCompoundRequestStore.getSelectedValue(
+                    group,
+                    index,
+                  )}
+                />
+              </div>
+            ))}
+          </div>
         </div>
+        {hasErrors && (
```

Comment:
What if we create a variable like `const requestBlockNumber = +item[0]` and will use it instead of hasErrors, useState and useEffect?